src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -1782,6 +1789,24 @@ private boolean shouldRedirectRelativeLinks(Element element) {
                     &&  currentPageElement != utils.getEnclosingTypeElement(element));
     }
 
+    /**
+     * Returns the output for an invalid tag. The returned content uses special styling to
+     * highlight the problem. Depending on the presence of the {@code detail} string the method
+     * returns a plain text span or an expandable component.
+     *
+     * @param summary the single-line summary message
+     * @param detail the optional detail message which may contain preformatted text
+     * @return the output
+     */
+    protected Content invalidTagOutput(String summary, Optional<Content> detail) {
+        if (detail.isEmpty() || detail.get().isEmpty()) {
+            return HtmlTree.SPAN(HtmlStyle.invalidTag, Text.of(summary));
+        }
+        return new HtmlTree(TagName.DETAILS).addStyle(HtmlStyle.invalidTag)
+                .add(new HtmlTree(TagName.SUMMARY).add(Text.of(summary)))
+                .add(new HtmlTree(TagName.PRE).add(detail.get()));
```

Comment:
For future cleanup (not now), consider adding factory methods for these constructs.

For example, consider:
* HtmlTree.PRE(String) or HtmlTree.PRE(Content...)
* HtmlTree.DETAILS(String summary, Content... body)